File: backend/app/schemas/dialog.py
from pydantic import BaseModel, field_validator
from datetime import datetime
from typing import Optional, Dict, Any


class DialogBase(BaseModel):
    """Base dialog schema"""
    dialog_type: str  # educational, test, assessment, reflective
    topic: Optional[str] = None

    @field_validator('dialog_type')
    @classmethod
    def dialog_type_valid(cls, v):
        allowed_types = ['educational', 'test', 'assessment', 'reflective']
        if not v or not v.strip():
            raise ValueError('Dialog type cannot be empty')
        if v not in allowed_types:
            raise ValueError(f'Dialog type must be one of: {", ".join(allowed_types)}')
        return v


class DialogCreate(DialogBase):
    """Schema for creating a dialog"""
    user_id: int


class DialogResponse(DialogBase):
    """Schema for dialog response"""
    dialog_id: int
    user_id: int
    started_at: datetime
    ended_at: Optional[datetime] = None
    extra_data: Dict[str, Any] = {}

    class Config:
        from_attributes = True

File: frontend/src/types/dialog.ts
/**
 * Dialog-related TypeScript types
 * Maps to backend schemas: @backend/app/schemas/dialog.py and @backend/app/schemas/message.py
 */

/**
 * Dialog type enumeration
 * @backend/app/schemas/dialog.py:13-18
 */
export type DialogType = 'educational' | 'test' | 'assessment' | 'reflective';

/**
 * Dialog interface matching backend DialogResponse schema
 * @backend/app/schemas/dialog.py:DialogResponse
 */
export interface Dialog {
  dialog_id: number;
  user_id: number;
  dialog_type: DialogType;
  topic: string | null;
  started_at: string;
  ended_at: string | null;
  extra_data: Record<string, any>;
}

/**
 * Sender type enumeration
 * @backend/app/schemas/message.py:25-30
 */
export type SenderType = 'user' | 'system';

/**
 * Message interface matching backend MessageResponse schema
 * @backend/app/schemas/message.py:MessageResponse
 */
export interface Message {
  message_id: number;
  dialog_id: number;
  sender_type: SenderType;
  content: string;
  is_question: boolean;
  timestamp: string;
  extra_data: Record<string, any>;
}
